pennylane/templates/subroutines/fable.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+This module contains the template for the Fast Approximate BLock Encoding (FABLE) technique.
+"""
+import warnings
+import numpy as np
+import pennylane as qml
+from pennylane.templates.state_preparations.mottonen import compute_theta, gray_code
+from pennylane.operation import AnyWires, Operation
+from pennylane.wires import Wires
+
+
+class FABLE(Operation):
+    r"""
+    Construct a unitary with the fast approximate block encoding method.
+
+    The FABLE method allows to simplify block encoding circuits without reducing accuracy, for matrices of specific structure [`arXiv:2205.00081 <https://arxiv.org/abs/2205.00081>`_].
+
+
+    Args:
+        A (tensor_like): an :math:`(N \times N)` matrix to be encoded, where N should have dimension equal to 2^n where n is an integer
+        tol (float): rotation gates that have an angle value smaller than this tolerance are removed
+        id (str or None): string representing the operation (optional)
+
+    Raises:
+        ValueError: if the number of wires doesn't fit the dimensions of the matrix
+
+    **Example**
+
+    We can define a matrix and a block-encoding circuit as follows:
+
+    .. code-block:: python
+
+        A = np.array([[-0.51192128, -0.51192128,  0.6237114 ,  0.6237114 ],
+                [ 0.97041007,  0.97041007,  0.99999329,  0.99999329],
+                [ 0.82429855,  0.82429855,  0.98175843,  0.98175843],
+                [ 0.99675093,  0.99675093,  0.83514837,  0.83514837]])
+        ancilla = ["ancilla"]
+        s = int(np.log2(A.shape[0]))
+        wires_i = [f"i{index}" for index in range(s)]
+        wires_j = [f"j{index}" for index in range(s)]
+        wire_order = ancilla + wires_i[::-1] + wires_j[::-1]
+        dev = qml.device('default.qubit')
+        @qml.qnode(dev)
+        def example_circuit():
+            qml.FABLE(A, tol=0.01)
+            return qml.state()
+
+    We can see that :math:`A` has been block encoded in the matrix of the circuit:
+
+    >>> M = len(A) * qml.matrix(circuit, wire_order=wire_order)().real[0 : len(A), 0 : len(A)]
+    ... print(f"Block-encoded matrix:\n{M}", "\n")
+    Block-encoded matrix:
+    [[-0.51192128 -0.51192128  0.6237114   0.6237114 ]
+    [ 0.97041007  0.97041007  0.99999329  0.99999329]
+    [ 0.82429855  0.82429855  0.98175843  0.98175843]
+    [ 0.99675093  0.99675093  0.83514837  0.83514837]]
```

Comment:
Could you simplify this example. We want to make sure that doc-string examples are very simple and isolate the feature we are adding. Instead a simple 2x2 matrix with values 0.1, 0.2, 0.3, 0.4 or something simple. Also there are some excess lines like the definition of wires_i, wires_j which can be removed. It's okay to just hardcode the wire_order for this example.